### ISSUE TITLE
Documentation should be updated to reflect loss scaling for manual optimization + gradient accumulation

### ISSUE THREAD

**Issue by callumhay** [Lightning-AI/lightning#17957]
I think the documentation should be updated to reflect that the loss for manual optimization with gradient accumulation requires manual scaling with the accumulation factor 'N'. The [current doc](https://lightning.ai/docs/pytorch/stable/model/manual_optimization.html) is a bit confusing since it specifies the use of % N for the optimizer stepping and zeroing of gradients but implicitly says nothing about scaling the actual loss and instead leaves it to the reader to scrutinize what `self.compute_loss(batch)` actually does.

I think it would be more instructive to have the example code specifically call this out, e.g., `loss = self.my_loss_fn(batch) / N`.

_Originally posted by @callumhay in https://github.com/Lightning-AI/lightning/issues/12811#issuecomment-1614987379_
            

cc @borda

**Comment by awaelchli** [comment]
Thanks @callumhay! Would you like to send an update for this? Explicitly dividing by N in that example should do the trick.
